content/machine-controller/cloud-providers/_index.en.md

```diff
@@ -0,0 +1,397 @@
+region: "eu-west"
+# linode size
+type: "g6-standard-2"
+# enable backups for the linode
+backups: false
+# enable private networking for the linode
+private_networking: true
+# add the following tags to the linode
+tags:
+- "machine-controller"
+```
+
+## OpenNebula
+
+{{% notice info %}}
+This is a [community provider]({{< relref "../#community-providers" >}}).
+{{% /notice %}}
+
+`machine.spec.providerConfig.cloudProviderSpec`:
+
+```yaml
+# XML-RPC endpoint of your OpenNebula installation
+endpoint: ""
+# your OpenNebula username
+username: ""
+# your OpenNebula password
+password: ""
+
+# cpu (float64)
+cpu: 1
+# vcpu
+vcpu: 2
+# memory in MB
+memory: 1024
+
+# the name of the image to use, needs to be owned by the current user
+image: "Amazon Linux 2"
+# which datastore to use for the image
+datastore: ""
+# size of the disk in MB
+diskSize: 51200
+
+# network name, needs to be owned by the current user
+network: ""
+
+# whether to enable the VNC console
+enableVNC: true
+
+# optional key/value pairs to add to the VM template
+vmTemplateExtra:
+  # useful for e.g. setting the placement attributes as defined in https://docs.opennebula.io/6.4/management_and_operations/references/template.html#template-placement-section
+  SCHED_REQUIREMENTS: 'RACK="G4"'
+```
+
+## OpenStack
+
+`machine.spec.providerConfig.cloudProviderSpec`:
+
+```yaml
+# identity endpoint of your openstack installation
+identityEndpoint: ""
+# application Credential ID and Secret can be used in place of username, password, tenantName/tenantID, and domainName.
+# application credentials ID
+applicationCredentialID: ""
+# application credentials secret
+applicationCredentialSecret: ""
+# your openstack username
+username: ""
+# your openstack password
+password: ""
+# the openstack domain
+domainName: "default"
+# project name
+projectName: ""
+# project id
+projectID: ""
+# tenant name (deprecated, should use projectName)
+tenantName: ""
+# tenant Id (deprecated, should use projectID)
+tenantID: ""
+# image to use (currently only ubuntu is supported)
+image: "Ubuntu 18.04 amd64"
+# instance flavor
+flavor: ""
+# additional security groups.
+# a default security group will be created which node-to-node communication
+securityGroups:
+  - "external-ssh"
+# the name of the subnet to use
+subnet: ""
+# [not implemented] the floating ip pool to use. When set a floating ip will be assigned o the instance
+floatingIpPool: ""
+# the availability zone to create the instance in
+availabilityZone: ""
+# the region to operate in
+region: ""
+# the name of the network to use
+network: ""
+# compute microversion
+computeAPIVersion: ""
+# set trust-device-path flag for kubelet
+trustDevicePath: false
+# set root disk size
+rootDiskSizeGB: 50
+# set root disk volume type
+rootDiskVolumeType: ""
+# set node-volume-attach-limit flag for cloud-config
+nodeVolumeAttachLimit: 20
+# the list of tags you would like to attach to the instance
+tags:
+  tagKey: tagValue
+```
+
+## Scaleway
```

Comment:
Is Scaleway supposed to be a non-community provider?

Reply:
:shrug: